### ISSUE TITLE
conda install fails on datascience-notebook and descendants 

### ISSUE THREAD

**Issue by jashubbard** [jupyter/docker-stacks#834]
I'm trying to build a custom notebook based on the datascience-notebook, but conda appears to be broken starting with the datascience-notebook but also with pyspark-notebook.

A minimal example using the scipy-notebook
```text
FROM jupyter/scipy-notebook:41e066e5caa8

USER $NB_UID
RUN conda install wget
```

This works fine. But if I try to do the same on the datascience-notebook (or pyspark-notebook): 

```text
FROM jupyter/datascience-notebook:41e066e5caa8

USER $NB_UID
RUN conda install wget
```

I get the following error: 

```
>>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/opt/conda/lib/python3.7/site-packages/conda/exceptions.py", line 1002, in __call__
        return func(*args, **kwargs)
      File "/opt/conda/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/opt/conda/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/opt/conda/lib/python3.7/site-packages/conda/cli/main_install.py", line 20, in execute
        install(args, parser, 'install')
      File "/opt/conda/lib/python3.7/site-packages/conda/cli/install.py", line 257, in install
        force_reinstall=context.force_reinstall or context.force,
      File "/opt/conda/lib/python3.7/site-packages/conda/core/solve.py", line 107, in solve_for_transaction
        force_remove, force_reinstall)
      File "/opt/conda/lib/python3.7/site-packages/conda/core/solve.py", line 145, in solve_for_diff
        force_remove)
      File "/opt/conda/lib/python3.7/site-packages/conda/core/solve.py", line 206, in solve_final_state
        self.prefix, update_modifier, deps_modifier, prune, ignore_pinned, force_remove
      File "/opt/conda/lib/python3.7/site-packages/conda/core/solve.py", line 740, in __init__
        self.solution_precs = tuple(self.prefix_data.iter_records())
      File "/opt/conda/lib/python3.7/site-packages/conda/core/prefix_data.py", line 117, in iter_records
        return itervalues(self._prefix_records)
      File "/opt/conda/lib/python3.7/site-packages/conda/core/prefix_data.py", line 146, in _prefix_records
        return self.__prefix_records or self.load() or self.__prefix_records
      File "/opt/conda/lib/python3.7/site-packages/conda/core/prefix_data.py", line 68, in load
        self._load_single_record(join(_conda_meta_dir, meta_file))
      File "/opt/conda/lib/python3.7/site-packages/conda/core/prefix_data.py", line 150, in _load_single_record
        with open(prefix_record_json_path) as fh:
    FileNotFoundError: [Errno 2] No such file or directory: '/opt/conda/conda-meta/blas-1.1-openblas.json'

`$ /opt/conda/bin/conda install wget`

  environment variables:
                 CIO_TEST=<not set>
                CONDA_DIR=/opt/conda
               CONDA_ROOT=/opt/conda
            CONDA_VERSION=4.6.7
         JULIA_DEPOT_PATH=/opt/julia
                     PATH=/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin

▽
                          :/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : None
       user config file : /home/jovyan/.condarc
 populated config files : /opt/conda/.condarc
          conda version : 4.6.8
    conda-build version : not installed
         python version : 3.7.1.final.0
       base environment : /opt/conda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/conda/pkgs
                          /home/jovyan/.conda/pkgs
       envs directories : /opt/conda/envs
                          /home/jovyan/.conda/envs
               platform : linux-64
             user-agent : conda/4.6.8 requests/2.21.0 CPython/3.7.1 Linux/3.10.0-957.1.3.el7.x86_64 ubuntu/18.04.1 glibc/2.27
                UID:GID : 1000:100
             netrc file : None
           offline mode : False
```

Is there something I'm missing?

**Comment by jashubbard** [comment]
I answered my own question-- I was running this on a server with an older version of Docker (1.13.1), and none of the environment variables were passed through. So `$NB_UID` evaluated to '' .  I'm surprised that any of the jupyter stacks worked given this issue.

**Comment by parente** [comment]
What you're doing looks correct.

Related to the blas JSON error you received, it looks like https://github.com/jupyter/docker-stacks/wiki/scipy-notebook-41e066e5caa8 has blas 1.1 but https://github.com/jupyter/docker-stacks/wiki/datascience-notebook-41e066e5caa8 has blas 2.4. It's just a hunch, but it seems relevant. 

You might look at older build manifests on https://github.com/jupyter/docker-stacks/wiki to see if an older tag for datascience notebook contains blas 1.1 and works. If so, that would narrow down the culprit.

**Comment by jashubbard** [comment]
FWIW, I discovered the the problem was more pervasive and had nothing to do with configuration-- the server I'm trying to run on is a CentOs 7 system. It turns out that it doesn't have `d_type` turned on for the xfs filesystem (running `docker system info` revealed the issue). This causes really bizarre problems with the files on the docker images I was building (reference: https://linuxer.pro/2017/03/what-is-d_type-and-why-docker-overlayfs-need-it/). I realized everything would build fine on my laptop, and on a random AWS instance, but not on this one server. 

Closing this issue, but hopefully this will help someone out who's searching based on these random `File Not Found` errors. 

Thanks!
